qiskit/pulse/configuration.py

```diff
@@ -14,11 +14,51 @@
 Configurations for pulse experiments.
 """
 from typing import Dict, Union, Tuple, Optional
+import numpy as np
 
 from .channels import PulseChannel, DriveChannel, MeasureChannel
 from .exceptions import PulseError
 
 
+def _assert_nested_dict_equal(a, b):
```

Comment:
How would you do that without making the provider a (circular) dependency of terra?

I think this function is okay. Maybe a comment could be added that the only reason it is needed is because `ndarray` overrides `__eq__` in an unhelpful way? The other things the JSON encoder handles are `complex` and `ParameterExpression` which work okay with `__eq__`.

Reply:
Yes, basically I agree with you. Unfortunately likely Qiskit doesn't have control of Kernel/Discriminator format, so we should continuously maintain this function to sync with the provider. I was thinking it was bit easier to delegate everything to the provider, but from the dependency viewpoint this approach is bit tough. 

I think at some point we should define subclass and use something like [pydantic](https://docs.pydantic.dev/latest/) rather than this free-form dictionary. Of course not in this PR :)